File: pyg-manager-web/src/main/webapp/js/service/brandService.js
//定义服务
app.service("brandService",function($http){
	//查询所有
	this.findAll = function(){
		return $http.get("../brand/findAll");
	};
	
	//分页
	this.findPage = function(page,rows){
		return $http.get('../brand/findPage?page=' + page + '&rows='+ rows);
	};
	
	//添加
	this.add = function(entity){
		
		//定义方法名
		var methodName = "add";
		//判断如果id存在,是修改
		if(entity.id!=null){
			methodName="update";
		}		
		return $http.post("../brand/"+methodName,entity);
	};
	
	//id查询品牌
	this.findOne = function(id){
		return $http.get("../brand/findOne?id="+id);
	};
	
	//查询
	this.search = function(page,rows,searchEntity){
		return $http.post("../brand/search?page="+page+"&rows="+rows,searchEntity);
	};
	
	//删除
	this.dele = function(selectIds){
		return $http.get("../brand/dele?ids=" + selectIds);
	};
});

File: pyg-manager/pyg-manager-interface/src/main/java/top/pyg/manager/service/BrandService.java
package top.pyg.manager.service;

import java.util.List;
import java.util.Map;

import top.pyg.pojo.TbBrand;
import top.pyg.utils.PageResult;
import top.pyg.utils.PygResult;


public interface BrandService {
	
	/**
	 * 需求:查询所有品牌数据
	 */
	public List<TbBrand> findAll();
	/**
	 * 需求:分页展示品牌列表
	 * 参数:Integer page,Integer rows
	 * 返回值:分页包装类对象PageResult
	 */
	public PageResult findPage(Integer page,Integer rows);

	//实现品牌增加
	public PygResult add(TbBrand brand);
	
	//查找一个品牌的信息
	public TbBrand findOne(Long id);
	
	//实现品牌信息修改
	public PygResult update(TbBrand brand);

	//实现品牌条件查询  pageNum当前页   pageSize当前页记录数
	public PageResult findBrandByPage(TbBrand brand,Integer page,Integer rows);
	
	//删除品牌信息
	public PygResult delete(Long[] ids);
	
	/**
	 * 品牌下拉框
	 */
	public List<Map> selectOptionList();
	
}
